python/tank/deploy/tank_commands/validate_config.py

```diff
@@ -46,36 +103,112 @@ def run_interactive(self, log, args):
         :param log: std python logger
         :param args: command line args
         """
-        if len(args) != 0:
-            raise TankError("This command takes no arguments!")
-        return self._run(log)
+
+        self._interactive = True
+
+        parser = argparse.ArgumentParser(
+            prog="./tank validate",
+            description="Environment validation command."
+        )
+
+        # a list of environment names
+        parser.add_argument(
+            "envs",
+            metavar="<env_name>",
+            type=str,
+            nargs="*",
+            help=self.parameters["envs"]["description"],
+        )
+
+        # mutually exclusive dump types, full and sparse
+        dump_type_group = parser.add_mutually_exclusive_group()
+        dump_type_group.add_argument(
+            "-f", "--dump",
+            action="store_true",
+            help=self.parameters["dump"]["description"],
+        )
+        dump_type_group.add_argument(
+            "-s", "--dump-sparse",
+            action="store_true",
+            help=self.parameters["dump_sparse"]["description"],
+        )
+
+        # the output directory for the modified environment files
+        parser.add_argument(
+            "-l", "--dump-location",
+            metavar="<output_directory>",
+            type=str,
+            help=self.parameters["dump_sparse"]["description"],
+        )
+
+        # turn on debug comments in the dumped environments
+        parser.add_argument(
```

Comment:
Please chat to @jfboismenu about this - he has done something very similar. One of these solutions should be moved up to the base class level I think.